iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -295,6 +295,64 @@ struct DotOpConversion
 };
 }  // namespace
 
+//===----------------------------------------------------------------------===//
+// mhlo.dot_general conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Converts mhlo.dot_general operation to linalg.batchmatmul op
+struct DotGeneralOpConversion
+    : public ConvertToLinalgBufferOp<DotGeneralOpConversion,
+                                     mhlo::DotGeneralOp> {
+  using ConvertToLinalgBufferOp<DotGeneralOpConversion,
+                                mhlo::DotGeneralOp>::ConvertToLinalgBufferOp;
+  LogicalResult apply(mhlo::DotGeneralOp op, ArrayRef<Value> inputBuffers,
+                      ArrayRef<Value> resultBuffers,
+                      ConversionPatternRewriter &rewriter) const {
+    auto extract1DVector = [](DenseIntElementsAttr elements) {
+      SmallVector<int64_t, 6> ret;
+      for (const APInt &element : elements) {
+        ret.push_back(element.getLimitedValue());
+      }
+      return ret;
+    };
+    mhlo::DotDimensionNumbers dimNumbers = op.dot_dimension_numbers();
+    auto lhsBatchingDims =
+        extract1DVector(dimNumbers.lhs_batching_dimensions());
```

Comment:
I think you can do
```
llvm::to_vector<6>(
  llvm::map_range(
    dimNumbers.lhs_batching_dimensions().getvalues<APInt>(),
    [](APInt v) { return v.getSExtValue(); }))
```

Reply:
I think the current way is more readable because the follows are using the same way. It's easier to see they are all doing the same thing, ie, calling extract1DVector.

Reply:
Ok, ill leave it upto you. What I suggested is the convention in MLIR.